packages/subgraph/src/dao/dao_v1_0_0.ts

```diff
@@ -164,49 +163,52 @@ export function handleNativeTokenDeposited(event: NativeTokenDeposited): void {
 }
 
 export function handleGranted(event: Granted): void {
-  // ContractPermissionId
-  let daoId = event.address.toHexString();
-  let contractPermissionIdEntityId =
-    event.params.where.toHexString() +
-    '_' +
-    event.params.permissionId.toHexString();
-  let contractPermissionIdEntity = ContractPermissionId.load(
-    contractPermissionIdEntityId
-  );
-  if (!contractPermissionIdEntity) {
-    contractPermissionIdEntity = new ContractPermissionId(
-      contractPermissionIdEntityId
-    );
-    contractPermissionIdEntity.dao = daoId;
-    contractPermissionIdEntity.where = event.params.where;
-    contractPermissionIdEntity.permissionId = event.params.permissionId;
-    contractPermissionIdEntity.save();
-  }
+  const contractAddress = event.address.toHexString();
+  const where = event.params.where;
+  const contractPermissionId = event.params.permissionId;
+  const who = event.params.who;
+
+  const permissionEntityId = [
+    contractAddress,
+    where.toHexString(),
+    contractPermissionId.toHexString(),
+    who.toHexString()
+  ].join('_');
+
+  const daoAddress = contractAddress;
 
   // Permission
-  let permissionId =
-    contractPermissionIdEntityId + '_' + event.params.who.toHexString();
-  let permissionEntity = new Permission(permissionId);
-  permissionEntity.dao = daoId;
-  permissionEntity.contractPermissionId = contractPermissionIdEntityId;
-  permissionEntity.where = event.params.where;
-  permissionEntity.who = event.params.who;
-  permissionEntity.actor = event.params.here;
-  permissionEntity.condition = event.params.condition;
-  permissionEntity.save();
+  let permissionEntity = Permission.load(permissionEntityId);
+  if (!permissionEntity) {
+    permissionEntity = new Permission(permissionEntityId);
+    permissionEntity.where = event.params.where;
+    permissionEntity.permissionId = contractPermissionId;
+    permissionEntity.who = event.params.who;
+    permissionEntity.actor = event.params.here;
+    permissionEntity.condition = event.params.condition;
+
+    permissionEntity.dao = daoAddress;
+    permissionEntity.save();
+  }
 }
 
 export function handleRevoked(event: Revoked): void {
   // permission
-  let permissionId =
-    event.params.where.toHexString() +
-    '_' +
-    event.params.permissionId.toHexString() +
-    '_' +
-    event.params.who.toHexString();
-  let permissionEntity = Permission.load(permissionId);
+  let contractAddress = event.address.toHexString();
+  let where = event.params.where;
+  let contractPermissionId = event.params.permissionId;
+  let who = event.params.who;
```

Comment:
Why are these and following variables `let` when they can be `const` as in `handleGranted`?

Reply:
amended